lib/order.rb

```diff
@@ -0,0 +1,77 @@
+require_relative 'menu'
+require_relative 'text_message'
+
+class Order
+
+  attr_reader :basket, :menu
+
+  def initialize(menu = Menu.new)
+    @basket = Hash.new(0)
+    @menu = menu
+  end
+
+  def add_dish(item)
+    @menu.check_availability(item)
+    @basket[item] += @menu.dishes[item]
+  end
+
+  def remove_dish(item)
+    check_basket(item)
+    @basket[item] -= @menu.dishes[item]
+  end
+
+  def view_basket
+    check_empty_basket
+    print_subtotals
+    print_total
+  end
+
+  def place_order(payment, confirmation = TextMessage.new)
+    check_empty_basket
+    review_payment(payment)
+    thank_you(payment, confirmation)
+  end
+
+  private
+
+  def check_basket(item)
+    msg = "Item not in the basket!"
+    raise msg if @basket[item] <= 0 
+  end
+
+  def print_subtotals
+    @basket.each do |item, price| 
+      if price.positive?
+        qty = (@basket[item] / @menu.dishes[item]).round
+        puts "#{qty}x #{item} £#{price}"
+      end
+    end
+  end
+
+  def checkout_total
+    @basket.select { |_, price| price.positive? }.values.reduce(:+)
+  end
+
+  def print_total
+    puts "----------------"
+    puts "Order Total £#{checkout_total}"
+  end
+  
+  def empty_basket?
```

Comment:
These methods could probably be set as private, as they're not meant to be called from the outside of the class

Reply:
Those methods are actually private :) The only public methods in the Order class are: add_dish, remove_dish, view_basket, place_order :)

Reply:
Oh yes sorry I completely missed the `private` keyword above 😅